test/connectivity.jl

```diff
@@ -256,6 +256,12 @@
     @test @inferred(!isgraphical([1, 1, 1]))
     @test @inferred(isgraphical([2, 2, 2]))
     @test @inferred(isgraphical(fill(3, 10)))
+
+    # Test directed simple graphicality
+    sdg = SimpleDiGraph(10, 90)
+    @test @inferred(isdigraphical(indegree(sdg), outdegree(sdg)))
+    @test !@inferred(isdigraphical([1,1,1], [1,1,0]))
+
```

Comment:
Can we also have a few edge cases here? Such as
- Empty vectors
- Graphs that only have a realization if there is a self-loop.
- Graphs where the sum of the in and out degree sequences if the same, but there is still no valid realization.

Reply:
We've just noticed that `isgraphical` and `isdigraphical` don't work properly in the edge cases you mentioned. 

For example: 
- `isgraphical(Integer[])` returns `true` even if it's not possible to realise a **simple undirected** graph (no self-loops, no multi-edges). It requires a check such as `!isempty(degs) || return false`; 
- `isdigraphical(Integer[], Integer[])` returns `true` even if it's not possible to realise a **simple directed** graph (no self-loops, no multi-edges). It requires a check such as `!(isempty(indegree_sequence) || isempty(indegree_sequence)) || return false`; 
- `isgraphical([2])` returns `true` even if it's not possible to realise a **simple undirected** graph (no self-loops, no multi-edges) with just a vertex with $d(v)=2$.
- `isdigraphical([1],[1])` returns `true` even if it's not possible to realise a **simple directed** graph (no self-loops, no multi-edges) with just a vertex with $d_{in}(v)=d_{out}(v)=1$.